### PR TITLE
Refactor utils.c and provide needed header.

### PR DESCRIPTION
**Pull request by movrsi**
Clean up attach.c to mark dump_desc as a debug feature.

### CODE REVIEW

File: ps2-decompile/utils.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#include "shared.h"
+#include <stdio.h>
+#include <usbd.h>
+
+int data_transfer(shared_t *g);
+
+#ifdef DEBUG
+UsbDeviceDescriptor *dump_desc(int devId);
+#endif
+#endif
```
Comment (Tatsh) [inline ps2-decompile/utils.h:14]:
```suggestion
#endif // UTILS_H
```

File: ps2-decompile/utils.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#ifndef UTILS_H
+#define UTILS_H
+
+#include "callbacks.h"
+#include "shared.h"
+#include <stdio.h>
+#include <usbd.h>
```
Comment (Tatsh) [inline ps2-decompile/utils.h:7]:
```suggestion
#include <stdio.h>

#include <usbd.h>

#include "callbacks.h"
#include "shared.h"
```

I want header order to be standard, 3rd party, own. They need a space between so clang-format will leave them.

File: ps2-decompile/utils.c [hunk 3]
```diff
@@ -1,37 +1,33 @@
-#include "callbacks.h"
-#include "shared.h"
-#include <stdio.h>
-#include <usbd.h>
+#include "utils.h"
 
 int data_transfer(shared_t *g) {
     int result; // $v0
+    // sceUsbdInterruptTransfer(g->c_pipe2, ?, g->maxPacketSize, data_transfer_done, g)
+    result = sceUsbdTransferPipe(g->dataEndpoint, &g->unk8[0x7804], g->maxPacketSize, 0,
```
Comment (Tatsh) [inline ps2-decompile/utils.c:6]:
```suggestion
    result = sceUsbdInterruptTransfer(g->dataEndpoint, &g->unk8[0x7804], g->maxPacketSize, data_transfer_done, g);
```

**Review by Tatsh — CHANGES_REQUESTED** [review]

**Review by Tatsh — CHANGES_REQUESTED** [review]

**Comment by movrsi** [comment]
Committed the request changes, let me know if you find anything else that may seem off.

**Pull request merged**